beater/cloudbeat.go

```diff
@@ -174,20 +179,19 @@ func (bt *cloudbeat) Run(b *beat.Beat) error {
 			if err := bt.configUpdate(update); err != nil {
 				bt.log.Errorf("Failed to update cloudbeat config: %v", err)
 			}
-
-		case fetchedResources := <-output:
-			cycleId, _ := uuid.NewV4()
-			cycleStart := time.Now()
-			bt.log.Infof("Eval cycle %v has started", cycleId)
-
-			cycleMetadata := transformer.CycleMetadata{CycleId: cycleId}
-			// TODO: send events through a channel and publish them by a configured threshold & time
-			events := bt.transformer.ProcessAggregatedResources(fetchedResources, cycleMetadata)
-
-			bt.log.Infof("Publishing %d events to index", len(events))
-			bt.client.PublishAll(events)
-
-			bt.log.Infof("Eval cycle %v published %d events after %s", cycleId, len(events), time.Since(cycleStart))
+		// Flush events to ES after a pre-defined interval, meant to clean residuals after a cycle is finished.
+		case <-time.Tick(flushInterval * time.Second):
+			logp.L().Infof("Publish cloudbeat events to elasticsearch after %d seconds", flushInterval)
+			bt.client.PublishAll(eventsToSend)
+			eventsToSend = nil
```

Comment:
not sure this is what you want. appending on nil later will fail. what about using make(..)?

Reply:
Actually, we can still append and iterate a Nil Slice even though its value is nil.
https://go.dev/play/p/jm8p0ErGaqA

Reply:
cool!